test/saramatest/sarama_group_session.go

```diff
@@ -50,5 +60,13 @@ func (gs *MockConsumerGroupSession) MessageWasMarked(msg *sarama.ConsumerMessage
 }
 
 func (gs *MockConsumerGroupSession) Context() context.Context {
-	return context.Background()
+	gs.RLock()
+	defer gs.RUnlock()
+	return gs.ctx
+}
+
+func (gs *MockConsumerGroupSession) SetContext(ctx context.Context) {
```

Comment:
where is this one called?

Reply:
by implementers, if they don't want to do a full produce/consume cycle from kafka they can fake it by using these mock types

Reply:
ah nice